spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/caffe/LayerConverter.scala

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.utils.caffe
+
+import scala.collection.JavaConverters._
+import caffe.Caffe
+import caffe.Caffe.EltwiseParameter.EltwiseOp
+import caffe.Caffe.{BlobProto, PoolingParameter, _}
+import com.google.protobuf.GeneratedMessage
+import com.intel.analytics.bigdl.nn.Graph._
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Table
+
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+
+
+class LayerConverter[T: ClassTag](implicit ev: TensorNumeric[T]) extends Converter[T]{
+
+  override protected def fromCaffeConvolution(layer : GeneratedMessage) : Seq[ModuleNode[T]] = {
+    val name = getLayerName(layer)
+    val param = getConvolutionParam(layer).get
+    val group = if (param.getGroup == 0)  1 else param.getGroup
```

Comment:
it couldn't be , all layer parameters are from the layer definition, it it's none, Caffe itself won't work